### PR TITLE
[Inflector] remove "internal" marker from the component

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | https://github.com/symfony/symfony/issues/26903
| License       | MIT
| Doc PR        | -

This code works: making it non-internal won't increase the maintenance burden and can help others build on it.

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
ping @mbabker

**Review by fabpot — APPROVED** [review]
... once the tests pass

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

File: src/Symfony/Component/Inflector/Tests/InflectorTest.php [hunk 1]
```diff
@@ -58,7 +58,7 @@ public function singularizeProvider()
             ['crises', ['cris', 'crise', 'crisis']],
             ['criteria', ['criterion', 'criterium']],
             ['cups', 'cup'],
-            ['data', ['daton', 'datum']],
```
Comment (andrew-demb) [inline src/Symfony/Component/Inflector/Tests/InflectorTest.php:61]: Why was this test changed?

Previously issues with singular `datum` were considered as a "feature" - https://github.com/symfony/symfony/issues/17764#issuecomment-183864898

**Pull request merged**